pyqtgraph/colormap.py

```diff
@@ -68,7 +69,10 @@ def __init__(self, pos, color, mode=None):
         self.pos = np.array(pos)
         order = np.argsort(self.pos)
         self.pos = self.pos[order]
-        self.color = np.array(color)[order]
+        self.color = []
+        for color in np.array(color)[order]:
+            self.color.append(mkColor(color).getRgb())
+        self.color = np.array(self.color)
```

Comment:
not to get all code-golfey, but this should likely be done as a list comprehension

```python
self.color = np.array([mkColor(color).getRgb() for color in np.array(color)[order]])
```

Reason I'm preferential to this method is that under the proposed implementation, not only is it a bit slower (because list-comprehensions are amazing like that), but at one point in the code `self.color` is a list, but at another point in the code, `self.color` is a np.array

Reply:
Thinking more about this, you can get even better performance (this may not exactly quite right).

```python
self.color = np.apply_along_axis(lambda x: mkColor(x).getRgb(), 0, np.array(color)[order])
```